build-system/tasks/cherry-pick.js

```diff
@@ -0,0 +1,164 @@
+const {green, cyan, red, yellow} = require('ansi-colors');
+
+/**
+ * Executes a shell command, and logs an error message if the command fails.
+ *
+ * @param {string} cmd
+ * @param {string} msg
+ * @return {!Object}
+ */
+function execOrLog(cmd, msg) {
+  const result = getOutput(cmd);
+  if (result.status) {
+    log(yellow('ERROR:'), msg);
+    throw new Error(result.stderr);
+  }
+
+  return result;
+}
+
+/**
+ * Determines the name of the cherry-pick branch.
+ *
+ * @param {string} version
+ * @param {Array<string>} commits
+ * @return {string}
+ */
+function cherryPickBranchName(version, commits) {
+  return `amp-release-${Number(version) + commits.length}`;
+}
+
+/**
+ * Updates tags from the remote and creates a branch at the release commit.
+ *
+ * @param {string} ref
+ * @param {string} branch
+ * @param {string} remote
+ */
+function prepareBranch(ref, branch, remote) {
+  const {status} = getOutput(`git rev-parse ${ref}`);
+  // Skip fetching remote tags if the tag is already available.
+  if (status === 0) {
+    log(green('INFO:'), 'Identified ref', cyan(ref), 'in local repository');
+  } else {
+    log(green('INFO:'), 'Fetching latest tags from', cyan(remote));
+    execOrLog(
+      `git fetch ${remote} "refs/tags/*:refs/tags/*"`,
+      `Failed to fetch updated tags from remote ${cyan(remote)}`
+    );
+  }
+
+  execOrLog(
+    `git checkout -b ${branch} ${ref}`,
+    `Failed to checkout new branch at ref ${cyan(ref)}`
+  );
+}
+
+/**
+ * Cherry-picks a commit into a new branch. When the cherry-pick succeeds,
+ * returns `true`. In the event of a merge conflict, the cherry-pick is aborted
+ * and an error is thrown.
+ *
+ * @param {string} sha
+ */
+function performCherryPick(sha) {
+  try {
+    log(green('INFO:'), 'Cherry-picking commit', cyan(sha));
+    execOrLog(
+      `git cherry-pick -x ${sha}`,
+      `Failed to cherry-pick commit ${cyan(sha)}; aborting`
+    );
+  } catch (e) {
+    log(green('INFO:'), 'Aborting cherry-pick of commit', cyan(sha));
+    getOutput(`git cherry-pick --abort`);
+    throw e;
+  }
+}
+
+function cherryPick() {
+  const {push, remote = 'origin'} = argv;
+  const commits = (argv.commits || '').split(',').filter(Boolean);
+  const onto = String(argv.onto || '');
+
+  if (!commits.length) {
+    log(red('ERROR:'), 'Must provide commit list with --commits');
+    process.exitCode = 1;
+    return;
+  }
+  if (!onto) {
+    log(red('ERROR:'), 'Must provide 13-digit AMP version with --onto');
+    process.exitCode = 1;
+    return;
+  }
+  if (onto.length === 15) {
+    log(
+      yellow('WARNING:'),
+      'Expected a 13-digit AMP version but got a 15-digit RTV;',
+      'ignoring channel prefix'
+    );
+    // Be forgiving if someone provides a version instead of a full RTV.
+    onto = onto.substr(2);
+  }
+
+  const branch = cherryPickBranchName(onto, commits);
+
+  try {
+    prepareBranch(onto, branch, remote);
+    commits.forEach(performCherryPick);
```

Comment:
git cherry-pick supports multiple commits. Can we do that instead of 1 per commit?

Reply:
My thinking here was twofold. The first is that this approach enables a more useful error message, as opposed to dumping the full Git output when the multiple-commit CP fails. The second reason I opted for doing each commit separately is that, in a follow-up PR, I anticipate adding a flag to allow for "force/fake merginge", where we commit the cherry-pick with `<<<<<` merge markers and still push the branch. The code will be broken and won't build, but would offer the release on-duty the ability to pick up where it left off with a partially-complete cherry-pick by checking out the branch.

Reply:
Ahh okay sounds good.